bitnami/pgpool/4/debian-12/rootfs/opt/bitnami/scripts/libpgpool.sh

```diff
@@ -591,6 +591,35 @@ pgpool_generate_password_file() {
     fi
 }
 
+########################
+# Encrypts a password
+# Globals:
+#   PGPOOL_*
+# Arguments:
+#   $1 - password
+# Returns:
+#   None
+#########################
+pgpool_encrypt_password() {
+    local -r password="${1:?missing password}"
+
+    local -a password_encryption_cmd=("pg_md5")
+
+    if [[ "$PGPOOL_AUTHENTICATION_METHOD" = "scram-sha-256" ]]; then
+
+        if is_file_writable "$PGPOOLKEYFILE"; then
+            # Creating a PGPOOLKEYFILE as it is writeable
+            echo "$PGPOOL_AES_KEY" > "$PGPOOLKEYFILE"
+            # Fix permissions for PGPOOLKEYFILE
+            chmod 0600 "$PGPOOLKEYFILE"
+        fi
+        password_encryption_cmd=("pg_enc" "--key-file=${PGPOOLKEYFILE}")
+        debug_execute "${password_encryption_cmd[@]}" "$password" | grep -o -E "AES.+" | tr -d '\n'
```

Comment:
This logic is duplicated from `pgpool_generate_password_file`. Could you unify it in a separate function or merge them to avoid code duplication?